File: lib_local/gallery/src/main/java/com/yanzhenjie/album/widget/divider/Divider.java
package com.yanzhenjie.album.widget.divider;

import androidx.recyclerview.widget.RecyclerView;

/**
 * 列表分割线抽象基类
 * 继承自 RecyclerView.ItemDecoration
 * 定义：分割线的宽度、高度接口
 */
public abstract class Divider extends RecyclerView.ItemDecoration {

    /**
     * 获取分割线高度
     * @return 分割线高度
     */
    public abstract int getHeight();

    /**
     * 获取分割线宽度
     * @return 分割线宽度
     */
    public abstract int getWidth();

}

File: lib_local/gallery/src/main/java/com/example/album/widget/recyclerview/divider/Divider.kt
package com.example.album.widget.recyclerview.divider

import androidx.recyclerview.widget.RecyclerView.ItemDecoration

/**
 * 列表分割线抽象基类
 * 继承自 RecyclerView.ItemDecoration
 * 定义：分割线的宽度、高度接口
 */
abstract class Divider : ItemDecoration() {
    /**
     * 获取分割线高度
     * @return 分割线高度
     */
    abstract fun getHeight(): Int

    /**
     * 获取分割线宽度
     * @return 分割线宽度
     */
    abstract fun getWidth(): Int
}